### DISCUSSION TITLE
Question about PLR

### DISCUSSION THREAD

**Discussion by garkavem** [DoubleML/doubleml-for-py#218] [Q&A] [answered]
Hello! I am sorry if my question is a bit dumb, but I don't understand how do we fit g0(X) model. I understand how we fit l0, m0 because we basically know X, Y, D... But we don't know \theta so we don't know what is Y-D\theta, no?
![dml_question](https://github.com/DoubleML/doubleml-for-py/assets/33484321/7113e9c7-a23a-41f5-a9f9-6fe25000403a)
(screenshot from DoubleML -- An Object-Oriented Implementation of Double Machine Learning in R)

**Comment by garkavem** [comment]
Also I wonder if I understand correctly, solving equation (12) from the same paper is like setting psi(W, \theta, \eta) = Y - l(x) - \theta * (D - m(x)), but then on page 10 equation (17) is this expression multiplied by (D - m(x)) ?..

**Reply by SvenKlaassen** [reply]
In equation (12), the model is defined. This is not equal to the equation, we are solving.
![image](https://github.com/DoubleML/doubleml-for-py/assets/47529404/d03cbbff-4967-404e-b5d9-2e776a9d7c90)
The equation we are solving is defined via the regression of $\hat{W}$ on $\hat{V}$ (see step (2))
![image](https://github.com/DoubleML/doubleml-for-py/assets/47529404/e27beac5-91c7-44a2-bcdc-f456588f8a6f)

By regressing $\hat{W}$ on $\hat{V}$, we are trying to minimize the MSE (as fitting a univariate linear model):

$$
min_{\theta} \mathbb{E}\big[(\hat{W} - \theta\hat{V})^2\big]
$$

This defines the equation via the first derivative:

$$
\mathbb{E}\big[(\hat{W} - \theta\hat{V})\hat{V}\big]
$$

which corresponds to equation (17) (by the definition of $\hat{W}$ and $\hat{V}$).

**Reply by garkavem** [reply]
Ah yes, indeed, thank you!

**Comment by SvenKlaassen** [comment, marked as answer]
This is a good question. 
You are correct, that it is quite hard to fit $g_0$ directly. The score is motivated from linear models where one would be able to obtain $g_0$ just via a regression of $Y$ on $D$ and $X$ (and then just use the prediction part from $X$). 
As most learners do not have this additive structure, we have to rely on a two-step procedure.
First a PLR model with "partialling-out" score is fitted to obtain a preliminary estimate $\tilde{\theta}$. Afterwards, we compute $g_0$ via regression of $Y-\tilde{\theta}D$ on $X$. This is then used to obtain the final estimate $\hat{\theta}$.

**Reply by garkavem** [reply]
Thank you very much it is much clearer now!

**Comment by garkavem** [comment]
@SvenKlaassen I am sorry for asking so many question, but I am just curious, in the lecture slides from [Tutorial on (Auto) DML based on Riesz Representers](https://www.chamberlainseminar.org/past-seminars/spring-2022#h.6kzhayrg1qra) on the slide 18 if I understand correctly we have "a recipe" of getting $\psi$... For PLR should we be able to obtain eq. (17) or (18) (from the beforementioned paper) from it and if yes, how? 
![image](https://github.com/DoubleML/doubleml-for-py/assets/33484321/fccba998-f8da-4c29-a755-a810fae92bbc)

**Reply by SvenKlaassen** [reply]
Yes that is correct. This is the IV-type score.
If you check Slide 9 in the presentation you can see that for the PLR model (I use the notation from the presentation):

$$
\begin{align*}
g(W)&=\theta D + f(X,A) \\
\alpha(W)&=\frac{D- \mathbb{E}[D|X,A]}{\mathbb{E}[(D- \mathbb{E}[D|X,A])^2]}
\end{align*}
$$

Further, we have

$$
m(W,g) = g(1, X, A) - g(0,X,A) = \theta
$$

such that

$$
\psi_\theta(Z;\theta;\alpha,g) = (Y - D\theta - f(X,A) ) \frac{D- \mathbb{E}[D|X,A]}{\mathbb{E}[(D- \mathbb{E}[D|X,A])^2]}
$$

Please remark that our score (see [documentation](https://docs.doubleml.org/dev/guide/scores.html#partially-linear-regression-model-plr)) is defined slightly differently

$$
\psi_{IV}(Z;\theta;\alpha,g) = (Y - D\theta - f(X,A) ) (D- \mathbb{E}[D|X,A])
$$

as we solve $\mathbb{E}[\psi(\dots)]=0$. This implies, we are not using the complete Riez Representer $\alpha(W)$ but have to fit $\mathbb{E}[D|X,A]$ with ML methods.

**Reply by garkavem** [reply]
Thanks a lot for the explanation!
